libs/types-common/src/Data/Domain.hs

```diff
@@ -20,20 +37,34 @@ domainText :: Domain -> Text
 domainText = _domainText
 
 mkDomain :: Text -> Either String Domain
-mkDomain =
-  bimap show Domain . Text.E.decodeUtf8'
-    <=< validateDomain . Text.E.encodeUtf8
-  where
-    -- this is a slightly hacky way of validating a domain,
-    -- but Text.Email.Validate doesn't expose the parser for the domain.
-    validateDomain = fmap Email.domainPart . Email.validate . ("local-part@" <>)
+mkDomain = Atto.parseOnly (domainParser <* Atto.endOfInput) . Text.E.encodeUtf8
 
 instance FromByteString Domain where
-  parser = do
-    bs <- Atto.takeByteString
-    case mkDomain =<< first show (Text.E.decodeUtf8' bs) of
-      Left err -> fail ("Failed parsing ByteString as Domain: " <> err)
-      Right domain -> pure domain
+  parser = domainParser
+
+domainParser :: Atto.Parser Domain
+domainParser = do
```

Comment:
heh, you couldn't help yourself.  :)

(i assume this is because the domain validation in the email package wasn't as thorough as this one.)

Reply:
Yes, it allows some extra things like comments (`(comment)example.com`) and bracketed IP addresses (`[1.1.1.1.]`).

Reply:
We now also decided to stick to FQDNs, requiring at least two labels, so I'll update the parser again.

Reply:
done.